workbench-core/accounts/src/utilities/hostingAccountLifecycleService.ts

```diff
@@ -203,7 +209,18 @@ export default class HostingAccountLifecycleService {
     };
 
     // Update bucket policy
-    await this._aws.clients.s3.putBucketPolicy(putPolicyParams);
+    try {
+      await this._aws.clients.s3.putBucketPolicy(putPolicyParams);
+    } catch (e) {
+      if (
+        e.name === 'MalformedPolicy' &&
```

Comment:
might want to pull the condition into a variable called `isInvalidAccountId` or something to make it easier to read whats happening here.

Reply:
Hmm, would that make it easier to read? I think the if statement with the 3 conditions are pretty self explanatory. I'm glad to make it a variable if you strongly feel it needs to be though.

`if (malformedPolicy && detail is related to arn with awsAccountId && is S3ServiceException) throw error`.